src/usb.c

```diff
@@ -777,37 +777,47 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, bool reset, char serial[16
 
     while (cnt--) {
         libusb_get_device_descriptor( list[cnt], &desc );
-        if (desc.idVendor != STLINK_USB_VID_ST)
-            continue;
-
         if (devBus && devAddr) {
             if ((libusb_get_bus_number(list[cnt]) != devBus)
                 || (libusb_get_device_address(list[cnt]) != devAddr)) {
                 continue;
             }
+            if (desc.idVendor != STLINK_USB_VID_ST) {
```

Comment:
devBus and devAddr is set when you declare STLINK_DEVICE environment variable.
So, if you set it and made a mistake then you get nothing useful in output.
Only one line will be printed (it a little bit confused me at the beginning):
```
$ STLINK_DEVICE=007:003 ./st-util
2017-02-18T23:32:14 INFO /home/user/repos/stlink/src/usb.c: bus 007 dev 003
$
```